### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by Abdulwhbhussain**
[Description](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md#description)
In this project, you will add authorization to your app using [CanCanCan](https://github.com/CanCanCommunity/cancancan).

[Project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md#project-requirements)
Install CanCanCan in your project.
Add a role column to the users table. Remember to use a migration for this.
A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @Abdulwhbhussain and @hetrox8,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Updating only the readme does not count as a contribution to the project and that is all one of you did. Kindly ensure everyone on the team commits real code.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,12 +2,27 @@ class PostsController < ApplicationController
...
+      @post.destroy
+      redirect_to user_posts_path(author)
+    else
+      @user = current_user
+      @post = Post.find(params[:id])
+      @post.destroy
+      redirect_to user_posts_path(@user)
+    end
+  end
+
```
Comment (andersonlebon) [inline app/controllers/posts_controller.rb]: - [x] Nice work on implementing the destroy logic,however the authorization should bemade with cancan, I can see you are manualy checking if the user the admin. You simply need to add the midlware `load_and_authorize_resource` which will be checking authomaticaly if the user is the Admin.

```rb
class PostsController < ApplicationController 
  `load_and_authorize_resource`
  def index
  end 
  ...
  ....
end

```

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Abdulwhbhussain  &&  @hetrox8 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is now professional :heavy_check_mark: 
- Nice use of cancan to make the authorization logic 👍 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Abdulwhbhussain) [reply app/controllers/posts_controller.rb]: Done Bro!

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @Abdulwhbhussain 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Add authorization rules
- ✔️ Limit access to web app resources based on authorization rules.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**